react/game/src/redux/auth/actions.js

```diff
@@ -1,76 +1,52 @@
+import { withPostSuccess, completeTypes, createTypes } from 'redux-recompose';
 import { service as authService, loadAuthState, saveAuthState, deleteAuthState } from '@services/authService';
 import { deleteGameState } from '@services/gameService';
 import { API } from '@config/api';
 import { push } from 'connected-react-router';
 import routes from '@constants/routes';
 
-export const AUTH_ACTIONS = {
-  LOAD_APP: 'LOAD_APP',
-  APP_LOADED: 'APP_LOADED',
-  LOGIN_LOADING: 'LOGIN_LOADING',
-  LOGIN_ERROR: 'LOGIN_ERROR',
-  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
-  LOGOUT: 'LOGOUT'
-};
+import { TARGET_APP_LOADING, TARGET } from './constants';
 
-const privateActionCreators = {
-  loadApp: ({ userId, token, email }) => ({
-    type: AUTH_ACTIONS.LOAD_APP,
-    payload: { userId, token, email }
-  }),
-  appLoaded: appLoaded => ({
-    type: AUTH_ACTIONS.APP_LOADED,
-    payload: { appLoaded }
-  }),
-  loginLoading: isLoading => ({
-    type: AUTH_ACTIONS.LOGIN_LOADING,
-    payload: { isLoading }
-  }),
-  loginError: errorMessage => ({
-    type: AUTH_ACTIONS.LOGIN_ERROR,
-    payload: { errorMessage }
-  }),
-  loginSuccess: (userId, token) => ({
-    type: AUTH_ACTIONS.LOGIN_SUCCESS,
-    payload: { userId, token }
-  }),
-  logout: () => ({
-    type: AUTH_ACTIONS.LOGOUT
-  })
-};
+const completedTypes = completeTypes(['INIT_APP_LOADING', 'LOGIN', 'LOGOUT'], ['INIT_APP_LOADING']);
+
+export const actions = createTypes(completedTypes, '@@AUTH');
 
 export const actionCreators = {
-  initApp: () => async dispatch => {
-    dispatch(privateActionCreators.appLoaded(false));
+  initApp: () => dispatch => {
     const session = loadAuthState();
```

Comment:
U need handle async all local storage services

Reply:
According to the HML5 API Docs, all the local storage calls are asynchronous, read this https://stackoverflow.com/questions/20231163/is-html5-localstorage-asynchronous